pytorch_lightning/loggers/base.py

```diff
@@ -22,25 +25,136 @@ def wrapped_fn(self, *args, **kwargs):
     return wrapped_fn
 
 
+def merge_dicts(
+        dicts: Sequence[Mapping],
+        agg_funcs: Optional[Mapping[str, Callable[[Sequence[float]], float]]] = None,
+        dflt_func: Callable[[Sequence[float]], float] = np.mean
+) -> Dict:
+    """Merge a sequence with dictionaries into one dictionary by aggregating the
+    same keys with some given function.
+
+    Args:
+        dicts:
+            Sequence of dictionaries to be merged.
+        agg_funcs:
+            Mapping from key name to function. This function will aggregate a
+            list of values, obtained from the same key of all dictionaries.
+            If some key has no specified aggregation function, the default one
+            will be used. Default is: None (all keys will be aggregated by the
+            default function).
+        dflt_func:
+            Default function to aggregate keys, which are not presented in the
+            `agg_funcs` map.
+
+    Returns:
+        Dictionary with merged values.
+
+    Examples:
+        >>> import pprint
+        >>> d1 = {'a': 1.7, 'b': 2.0, 'c': 1}
+        >>> d2 = {'a': 1.1, 'b': 2.2}
+        >>> d3 = {'a': 1.1, 'v': 2.3}
+        >>> dflt_func = min
+        >>> agg_funcs = {'a': np.mean, 'v': max}
+        >>> pprint.pprint(merge_dicts([d1, d2, d3], agg_funcs, dflt_func))
+        {'a': 1.3, 'b': 2.0, 'c': 1, 'v': 2.3}
+    """
+
+    keys = list(functools.reduce(operator.or_, [set(d.keys()) for d in dicts]))
+    dx = {}
+    for k in keys:
+        fn = agg_funcs.get(k, dflt_func) if agg_funcs else dflt_func
+        agg_val = fn([v for v in [d.get(k) for d in dicts] if v is not None])
+        dx[k] = agg_val
+
+    return dx
+
+
 class LightningLoggerBase(ABC):
     """Base class for experiment loggers."""
 
-    def __init__(self):
+    def __init__(
+            self,
+            agg_funcs: Optional[Mapping[str, Callable[[Sequence[float]], float]]] = None,
+            dflt_func: Callable[[Sequence[float]], float] = np.mean
```

Comment:
this shall be depending on metric, not one for all in single logger...

Reply:
Could you, clarify a little? Did not get your idea